hudi-utilities/src/main/java/org/apache/hudi/utilities/transform/Transformer.java

```diff
@@ -45,4 +46,8 @@ public interface Transformer {
    */
   @PublicAPIMethod(maturity = ApiMaturityLevel.STABLE)
   Dataset<Row> apply(JavaSparkContext jsc, SparkSession sparkSession, Dataset<Row> rowDataset, TypedProperties properties);
+
+  default Schema schemaTransform(JavaSparkContext jsc, SparkSession sparkSession, Schema incomingSchema, TypedProperties properties) {
+    throw new UnsupportedOperationException("Not implemented");
```

Comment:
lets throw a specific Hudi exception here?

Reply:
Should it return an `Option` instead? The handling of code in an upper layer need not be written in terms of exception handling then?

Reply:
Addressed

Reply:
@vinothchandar  and @lokeshj1703 if we return an empty option here, schema validation should fail, right? Validation should be strict or else you get a false positive telling you that your combination of transforms is valid when you actually don't have insight.

Reply:
Also thoughts on changing the signature to use StructType instead of avro schemas since transforms deal with Rows?

Reply:
Yes, we have the checks for making sure that Option is not empty if `enableSchemaValidation` is true.